lib/new_relic/agent/http_clients/async_http_wrappers.rb

```diff
@@ -0,0 +1,83 @@
+# This file is distributed under New Relic's license terms.
+# See https://github.com/newrelic/newrelic-ruby-agent/blob/main/LICENSE for complete details.
+# frozen_string_literal: true
+
+require_relative 'abstract'
+require 'resolv'
+
+module NewRelic
+  module Agent
+    module HTTPClients
+      class AsyncHTTPResponse < AbstractResponse
+        def get_status_code
+          get_status_code_using(:status)
+        end
+
+        def [](key)
+          to_hash[key.downcase]&.first
+        end
+
+        def to_hash
+          @wrapped_response.headers.to_h
+        end
+      end
+
+      class AsyncHTTPRequest < AbstractRequest
+        def initialize(connection, method, url, headers)
+          @connection = connection
+          @method = method
+          @url = ::NewRelic::Agent::HTTPClients::URIUtil.parse_and_normalize_url(url)
+          @headers = headers
+        end
+
+        ASYNC_HTTP = 'Async::HTTP'
+        LHOST = 'host'
+        UHOST = 'Host'
+        COLON = ':'
```

Comment:
Not for this PR, but maybe it's time to move these constants to a shared location.